### PR TITLE
About page

### PR DESCRIPTION
**Pull request by mdamonte88**
About page - As a user, I should be able to check the about section so I can learn more about Target MVD

### CODE REVIEW

**Comment by xlasserre** [comment]
@mdamonte88 Can you provide a description and screenshots to this PR? Also, you should assign reviewers, the more the better, so the PR is approved faster.

File: src/components/common/Menu.js [hunk 1]
```diff
@@ -1,16 +1,46 @@
...
+  static propTypes = {
+    show: PropTypes.bool,
+    intl: intlShape.isRequired,
+  };
 
+  render() {
+    const { show, intl } = this.props;
+
+    return (
+      <div className="containerMenu" style={{ display: show ? 'block' : 'none' }} >
```
Comment (felipegarcia92) [inline src/components/common/Menu.js]: If you just set display to none, you'd be rendering the the tags, but they're hidden.

The idea is that you don't render what you don't need. To not render, just return null. You can do something like:

`if (!show) return null;`

File: src/components/user/AboutTarget.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+// Imagenes
```
Comment (felipegarcia92) [inline src/components/user/AboutTarget.js]: Remove, not necessary.

File: src/components/user/MenuLeft.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+  render() {
+    const { topicList, section, title, handleCreateTarget, history } = this.props;
+    const { location } = history;
+
+    return section === 'newTarget' || section === 'aboutTarget' ?
+      (
+        <div className="slide slideLeft col-6">
+          <div className="headerContent">
+            <FormattedMessage id={title} /> <Link to={location && location.pathname === '/home' ? routes.index : routes.home} > <div className="closeIcon" /> </Link>
```
Comment (felipegarcia92) [inline src/components/user/MenuLeft.js]: Can you split this line? It's too long.

File: src/components/user/MenuLeft.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+          </div>
+          <div className="content create-target">
+            {section === 'newTarget' ? <CreateTargetForm onSubmit={handleCreateTarget} topics={topicList} /> : <AboutTarget /> }
```
Comment (felipegarcia92) [inline src/components/user/MenuLeft.js]: Same here.

File: src/styles/sign-up-page.scss [hunk 5]
```diff
@@ -226,6 +226,16 @@
       font-weight: bold;
       letter-spacing: 3px;
     }
+
+    .closeIcon {
```
Comment (felipegarcia92) [inline src/styles/sign-up-page.scss]:
```suggestion
    .close-icon {
```
Remember, all css class names should be written `in-this-format`

File: src/styles/sign-up-page.scss [hunk 6]
```diff
@@ -384,7 +394,33 @@
 }
 
 /*
-  #Login-Page
+  About-page
+*/
+
+.aboutpage {
```
Comment (felipegarcia92) [inline src/styles/sign-up-page.scss]: Same here

File: src/styles/sign-up-page.scss [hunk 7]
```diff
@@ -384,7 +394,33 @@
...
+      margin: 24px 14px;
+      width: calc(100% - 28px);
+    }
+
+    p {
+      text-align: justify;
+    }
+  }
+
+  .iconSmile {
```
Comment (felipegarcia92) [inline src/styles/sign-up-page.scss]: Same here.

**Review by felipegarcia92 — CHANGES_REQUESTED** [review]

File: src/components/common/Menu.js (hunk 1 above)
Comment (mdamonte88) [reply src/components/common/Menu.js]: It sounds good. I think is more performant.

File: src/components/user/AboutTarget.js (hunk 2 above)
Comment (mdamonte88) [reply src/components/user/AboutTarget.js]: It´s done!!

File: src/components/user/MenuLeft.js [hunk 8]
```diff
@@ -24,8 +24,8 @@ export default class MenuLeft extends PureComponent {
     return section === 'newTarget' || section === 'aboutTarget' ?
       (
         <div className="slide slideLeft col-6">
-          <div className="headerContent">
-            <FormattedMessage id={title} /> <Link to={location && location.pathname === '/home' ? routes.index : routes.home} > <div className="closeIcon" /> </Link>
+          <div className="header-content">
+            <FormattedMessage id={title} /> <Link to={location && location.pathname === '/home' ? routes.index : routes.home} > <div className="close-icon" /> </Link>
```
Comment (felipegarcia92) [inline src/components/user/MenuLeft.js]: Split this line and activate max line length on linter

Comment (mdamonte88) [reply src/components/user/MenuLeft.js]: It´s done!!

File: src/styles/sign-up-page.scss (hunk 5 above)
Comment (mdamonte88) [reply src/styles/sign-up-page.scss]: It´s done!!

File: src/components/user/MenuLeft.js (hunk 4 above)
Comment (felipegarcia92) [reply src/components/user/MenuLeft.js]: @mdamonte88 can you split this one too?

File: src/components/user/MenuLeft.js (hunk 3 above)
Comment (mdamonte88) [reply src/components/user/MenuLeft.js]: It´s done!!

**Review by felipegarcia92 — APPROVED** [review]

File: src/styles/styles.scss [hunk 9]
```diff
@@ -61,6 +61,12 @@ body {
     list-style-type: none;
     margin: 0;
 
+    li {
```
Comment (agch-dev) [inline src/styles/styles.scss:64]: Take advantage of sass
```
li {
  a {
    color: $black;
  }
  
  &.selected {
    font-weight: bold;
  }
}
```

File: src/styles/responsive-styles.scss [hunk 10]
```diff
@@ -23,6 +23,11 @@
     .slide[class*="col-"] {
       width: 100%;
     }
+    .slide.slideLeft[class*="col-"] {
```
Comment (agch-dev) [inline src/styles/responsive-styles.scss]: a lot of repetition that could be better handled if you take advantage of sass

File: src/containers/HomePage.js [hunk 11]
```diff
@@ -60,40 +58,25 @@ class HomePage extends PureComponent {
...
-      );
-  }
-
   render() {
-    const { targetList, topicList } = this.props;
+    const { targetList, topicList, history } = this.props;
+    const { isCreatingNewTarget } = this.state;
+    const { location } = history;
+    let section = isCreatingNewTarget ? 'newTarget' : 'welcome';
+    section = location.pathname === '/about' ? 'aboutTarget' : section;
```
Comment (agch-dev) [inline src/containers/HomePage.js]: you probably already have the `/about` route defined in routesPath right? perhaps use it from there?

File: src/components/user/MenuLeft.js [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+      (
+        <div className="slide slideLeft col-6">
+          <div className="header-content">
+            <FormattedMessage id={title} />
+            <Link to={location && location.pathname === '/home' ? routes.index : routes.home} >
+              <div className="close-icon" />
+            </Link>
+          </div>
+          <div className="content create-target">
+            {section === 'newTarget' ? (
```
Comment (agch-dev) [inline src/components/user/MenuLeft.js]: Storing the section types into a constants object might be a good idea

File: src/components/user/MenuLeft.js [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+  render() {
+    const { topicList, section, title, handleCreateTarget, history } = this.props;
+    const { location } = history;
+
+    return section === 'newTarget' || section === 'aboutTarget' ?
+      (
+        <div className="slide slideLeft col-6">
+          <div className="header-content">
+            <FormattedMessage id={title} />
+            <Link to={location && location.pathname === '/home' ? routes.index : routes.home} >
```
Comment (agch-dev) [inline src/components/user/MenuLeft.js]: same use routesPath value instead of string literal

**Review by agch-dev — COMMENTED** [review]
Great job! Just a few comments

File: src/components/user/MenuLeft.js (hunk 4 above)
Comment (mdamonte88) [reply src/components/user/MenuLeft.js]: It´s done!

File: src/styles/sign-up-page.scss (hunk 7 above)
Comment (mdamonte88) [reply src/styles/sign-up-page.scss]: It´s done!!

File: src/styles/styles.scss (hunk 9 above)
Comment (mdamonte88) [reply src/styles/styles.scss:64]: It sounds good!

File: src/containers/HomePage.js (hunk 11 above)
Comment (mdamonte88) [reply src/containers/HomePage.js]: I didn't want to use this, perhaps it looks better!!

Comment (mdamonte88) [reply src/containers/HomePage.js]: I replaced this by 
```
const showMenu = location.pathname === '/about' && !isCreatingNewTarget;
let section = isCreatingNewTarget ? 'newTarget' : 'welcome';
section = location.pathname === '/about' ? 'aboutTarget' : section;
```

File: src/components/common/Menu.js [hunk 14]
```diff
@@ -1,16 +1,46 @@
...
+        <div className="hamburgerIcon" />
+        <div className="close" />
+        <ul className="menuList">
+          <li className="selected">
+            <Link className="forgot-your-password" to={routes.about}>
+              {intl.formatMessage(messages.about)}
+            </Link>
+          </li>
+          <li>
+            <Link className="forgot-your-password" to={routes.contact}>
```
Comment (xlasserre) [inline src/components/common/Menu.js]: Here and on line 32 you are using the forgot-your-password class for blocks not related about the password. You should change the class name if you are going to use it.

File: src/components/user/MenuLeft.js [hunk 15]
```diff
@@ -0,0 +1,51 @@
...
+import CreateTargetForm from 'components/user/CreateTargetForm';
+import AboutTarget from 'components/user/AboutTarget';
+import routes from 'constants/routesPaths';
+
+export default class MenuLeft extends PureComponent {
+  static propTypes = {
+    title: string,
+    topicList: array,
+    section: string,
+    handleCreateTarget: func,
```
Comment (xlasserre) [inline src/components/user/MenuLeft.js]: This function should be required, right?

File: src/components/user/MenuLeft.js [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+              <AboutTarget />
+            )}
+          </div>
+        </div>
+      ) :
+      (
+        <div className="slide slideLeft col-6">
+          <Welcome currentPage="Home" />
+          <div className="content">
+            <LogoutButton className="sign-in-button" />
```
Comment (xlasserre) [inline src/components/user/MenuLeft.js:49]: Also, here you are using a class name that does not have anything to do with the block they are in.

File: src/components/user/TargetPoint.js [hunk 17]
```diff
@@ -6,12 +6,12 @@ import twitterIcon from './../../assets/socialNetworks/twitter@3x.png';
 import playVideoIcon from './../../assets/phone/play@3x.png';
 
 const TargetPoint = () => (
-  <div>
+  <div className="targetPoint">
```
Comment (xlasserre) [inline src/components/user/TargetPoint.js]: Class names should be lowercase and hyphenated.

File: src/locales/es.js [hunk 18]
```diff
@@ -18,10 +18,12 @@ export default {
   'login.form.email.invalid': 'Debe ingresar un email válido',
   'login.form.password.presence': 'Debe ingresar una contraseña para continuar',
 
+  // Menu
   'menu.about': 'ACERCA',
   'menu.contact': 'CONTACTO',
 
-  'signup.title': 'REGISTRARSE',
+  // SignUp
+  'signup.title': 'REGISRARSE',
```
Comment (xlasserre) [inline src/locales/es.js]: There is a missing 'T' here

**Review by xlasserre — APPROVED** [review]
Nice job! Left a few comments. And it would be a good idea for you to provide screenshots in the description of the PR so we can see what the end result looks like.

File: src/components/user/MenuLeft.js (hunk 13 above)
Comment (mdamonte88) [reply src/components/user/MenuLeft.js]: I changed this.

File: src/components/user/MenuLeft.js (hunk 12 above)
Comment (mdamonte88) [reply src/components/user/MenuLeft.js]: I changed this

**Review by xlasserre — CHANGES_REQUESTED** [review]
Sorry, I meant to request changes but checked approve.

File: src/actions/actionTypes.js [hunk 19]
```diff
@@ -9,3 +9,8 @@ export const ADD_TARGET_FAILED = 'ADD_TARGET_FAILED';
 // TOPICS
 export const LOAD_TOPICS_SUCCESS = 'LOAD_TOPICS_SUCCESS';
 export const LOAD_TOPICS_FAILED = 'LOAD_TOPICS_FAILED';
+
+// SECTIONS
+export const NEW_TARGET = 'NEW_TARGET';
```
Comment (agch-dev) [inline src/actions/actionTypes.js]: Hmm these are not actions, I was thinking something along the lines of the following and it should go in constants file.
```
export const sectionTypes = {
  newTarget: 'newTarget',
  aboutTaget: 'aboutTarget',
  welcome: 'welcome'
}
```

File: src/components/user/MenuLeft.js [hunk 20]
```diff
@@ -17,21 +18,23 @@ export default class MenuLeft extends PureComponent {
...
   };
 
+
   render() {
     const { topicList, section, title, handleCreateTarget, history } = this.props;
     const { location } = history;
+    const pathname = location && location.pathname;
 
-    return section === 'newTarget' || section === 'aboutTarget' ?
+    return section === sections.NEW_TARGET || section === sections.ABOUT_TARGET ?
```
Comment (agch-dev) [inline src/components/user/MenuLeft.js]: Once you import sections object you can destructure it.

File: src/containers/HomePage.js [hunk 21]
```diff
@@ -62,9 +63,9 @@ class HomePage extends PureComponent {
     const { targetList, topicList, history } = this.props;
     const { isCreatingNewTarget } = this.state;
     const { location } = history;
-    let section = isCreatingNewTarget ? 'newTarget' : 'welcome';
-    section = location.pathname === '/about' ? 'aboutTarget' : section;
-    const showMenu = section !== 'aboutTarget' && !isCreatingNewTarget;
+    const showMenu = location.pathname === '/about' && !isCreatingNewTarget;
```
Comment (agch-dev) [inline src/containers/HomePage.js]: still using string instead of route for `/about`

File: src/components/common/Menu.js (hunk 14 above)
Comment (mdamonte88) [reply src/components/common/Menu.js]: I removed this unnecessary class.

File: src/styles/responsive-styles.scss [hunk 22]
```diff
@@ -3,31 +3,35 @@
...
-    .slide[class*="col-"] {
-      width: 100%;
-    }
-    .slide.slideLeft[class*="col-"] {
-      width: 100%;
-    }
-    .slide.slideCenter[class*="col-"] {
-      width: 100%;
+    .slide {
+      &[class*="col-"] {
```
Comment (agch-dev) [inline src/styles/responsive-styles.scss]: Oh hadn't noticed that they all had the same property. Btw isn't  `&[class*="col-"]` enough to apply to all? If it's not enough perhaps it has to do with rule priorities. Anyhow, if it isn't you can still simplify this:
```
.slides[class*="col-"] {
  &,
  &.slideLeft,
  &.slideCenter {
    width: 100%;
  }
}
```
I think that `[class*="col-"]` can go before the `.slideLeft` and `.slideCenter`

If not this should work
```
.slides {
  &,
  &.slideLeft,
  &.slideCenter {
    &[class*="col-"] {
      width: 100%;
    }
  }
}
```

**Review by agch-dev — CHANGES_REQUESTED** [review]

File: src/components/user/MenuLeft.js (hunk 15 above)
Comment (mdamonte88) [reply src/components/user/MenuLeft.js]: In this case, the children component is which using this function and has a prop required. I added this in any way.

File: src/components/user/TargetPoint.js (hunk 17 above)
Comment (mdamonte88) [reply src/components/user/TargetPoint.js]: I forgot this.

File: src/locales/es.js (hunk 18 above)
Comment (mdamonte88) [reply src/locales/es.js]: Yes, it had fixed previously but in the merge, I didn't see

File: src/containers/HomePage.js (hunk 11 above)
Comment (agch-dev) [reply src/containers/HomePage.js]: hmm sorry for commenting the same again bellow. I had not seen this. Why didn't you want to do that?

File: src/components/user/MenuLeft.js (hunk 16 above)
Comment (xlasserre) [reply src/components/user/MenuLeft.js:49]: @mdamonte88 This one is still not solved

File: src/styles/sign-up-page.scss (hunk 7 above)
Comment (mdamonte88) [reply src/styles/sign-up-page.scss]: It's done!

File: src/styles/responsive-styles.scss (hunk 22 above)
Comment (mdamonte88) [reply src/styles/responsive-styles.scss]: I agree with you

**Review by agch-dev — COMMENTED** [review]
✅

**Pull request merged**